#! /bin/sh -e
## 20_unix_oss.dpatch by  <mikma@users.sourceforge.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Bugfix of Pa_StreamTime and Pa_UpdateStreamTime.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /var/local/src/portaudio/portaudio-18.1/pa_unix_oss/pa_unix_oss.c portaudio-18.1/pa_unix_oss/pa_unix_oss.c
--- /var/local/src/portaudio/portaudio-18.1/pa_unix_oss/pa_unix_oss.c	2003-06-30 11:05:50.000000000 +0200
+++ portaudio-18.1/pa_unix_oss/pa_unix_oss.c	2004-09-08 22:43:34.000000000 +0200
@@ -346,13 +346,13 @@
     if( pahsc->pahsc_NativeOutputBuffer )
     {
        ioctl(pahsc->pahsc_OutputHandle, SNDCTL_DSP_GETOPTR, &info);
-       delta = (info.bytes - pahsc->pahsc_LastPosPtr) & 0x000FFFFF;
+       delta = (info.bytes - pahsc->pahsc_LastPosPtr);
        return (pahsc->pahsc_LastStreamBytes + delta) / (past->past_NumOutputChannels * sizeof(short));
     }
     else
     {
        ioctl(pahsc->pahsc_InputHandle, SNDCTL_DSP_GETIPTR, &info);
-       delta = (info.bytes - pahsc->pahsc_LastPosPtr) & 0x000FFFFF;
+       delta = (info.bytes - pahsc->pahsc_LastPosPtr);
        return (pahsc->pahsc_LastStreamBytes + delta) / (past->past_NumInputChannels * sizeof(short));
     }
 }
@@ -372,7 +372,7 @@
   {
     ioctl(pahsc->pahsc_InputHandle, SNDCTL_DSP_GETIPTR, &info);
   }
-  delta = (info.bytes - pahsc->pahsc_LastPosPtr) & 0x000FFFFF;
+  delta = (info.bytes - pahsc->pahsc_LastPosPtr);
   pahsc->pahsc_LastStreamBytes += delta;
   pahsc->pahsc_LastPosPtr = info.bytes;
 }
